perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ConsolidatorManagerImpl.java

```diff
@@ -600,7 +692,20 @@ else if (ues.getExtSource().getName().equals("https://login.elixir-czech.org/idp
 						} catch (AttributeNotExistsException | WrongAttributeAssignmentException ex) {
 							// don't care
 						}
-						es.add(ues.getExtSource());
+						Attribute nameAttr = null;
+						try {
+							nameAttr = perun.getAttributesManagerBl().getAttribute(registrarSession, ues, AttributesManager.NS_UES_ATTR_DEF + ":sourceIdPName");
+						} catch (Exception e) {
+							log.debug("Failed to get sourceIdpName attribute" + e);
+						}
+						var richExtSource = new RichExtSource(ues.getExtSource());
```

Comment:
so the proposed solution to the problem with the enriched ext sources is to return List of Identities from this method that doesn't contain any additional attributes, create new object EnrichedIdentity which would contain EnrichedExtSource and a new method that would "enrich" the Identity object to create enriched one from it. this way a new method could be added checking similar users that would be declared to return List of EnrichedIdentity and would allow us to know that it contains the attributes inside the EnrichedExtSource property of the EnrichedIdentity. the new method would first call the old one and then enrich the identities.